flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphDeploymentTest.java

```diff
@@ -397,14 +404,16 @@ public void testRegistrationOfExecutionsCanceled() {
 			JobVertex v1 = new JobVertex("v1", jid1);
 			JobVertex v2 = new JobVertex("v2", jid2);
 
-			Map<ExecutionAttemptID, Execution> executions = setupExecution(v1, 19, v2, 37).f1;
+			Tuple2<ExecutionGraph, Map<ExecutionAttemptID, Execution>> graphExecutionsTuple = setupExecution(v1, 19, v2, 37);
+			ExecutionGraph testExecutionGraph = graphExecutionsTuple.f0;
+			Map<ExecutionAttemptID, Execution> executions = new ConcurrentHashMap<>(graphExecutionsTuple.f1);
```

Comment:
Why are you using `ConcurrentHashMap`? There won't be concurrent modifications as far as I am concerned. 
Moreover, we do not really require a Map, i.e., something like `Collection<Execution>` would suffice.

```suggestion
			Collection<Execution> executions = new ArrayList<>(graphExecutionsTuple.f1.values());
```

Reply:
The same holds for the other tests.

Reply:
You are right. I was working on solving the HashMap caused `ConcurrentModificationException` when iterating, so my mind was limited.

Reply:
I have updated the PR.